elasticreindexer/process/reindexerMultiWrites.go

```diff
@@ -0,0 +1,139 @@
+package process
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-tools-go/elasticreindexer/config"
+)
+
+type interval struct {
+	start int64
+	stop  int64
+}
+
+type reindexerMultiWrite struct {
+	indicesNoTimestamp   []string
+	indicesWithTimestamp []string
+	numParallelWrite     int
+	blockChainStartTime  int64
+	enabled              bool
+
+	reindexerClient *reindexer
+}
+
+func NewReindexerMultiWrite(cfg *config.GeneralConfig) (*reindexerMultiWrite, error) {
+	ri, err := CreateReindexer(cfg)
```

Comment:
We should have checks for the configs values

Reply:
you still use some config values when creating the reindexerMultiWrite object. You can add all of them into a private method checkArgs.

Also it may be worth having a DTO object so NewReindexerMultiWrite would receive only the necessary information, not the whole GeneralConfig.

Reply:
refactored